### PR TITLE
#1539 block processing of new claims if old ones are still in flight

### PR DESCRIPTION
**Pull request by krzyk**
#1539 
* New claims now won't be read from SQS if all the threads processing claims are still working

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1578](https://github.com/zerocracy/farm/issues/1578) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1578](https://github.com/zerocracy/farm/issues/1578) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2221245027); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/zerocracy/claims/ClaimsRoutine.java [hunk 1]
```diff
@@ -119,48 +128,52 @@ public void start(final ShutdownFarm.Hook shutdown) {
...
+                new ReceiveMessageRequest(queue)
+                    .withMessageAttributeNames(
+                        "project", "signature", ClaimsRoutine.UNTIL
+                    )
+                    .withMaxNumberOfMessages(ClaimsRoutine.LIMIT)
+            ).getMessages();
+            final Set<String> projects = new HashSet<>();
+            final List<Message> merged = new LinkedList<>();
+            for (final Message message : messages) {
+                final Map<String, MessageAttributeValue> attr =
```
[the remark this answers is not in the document]
Comment (krzyk) [reply src/main/java/com/zerocracy/claims/ClaimsRoutine.java]: @paulodamaso I didn't change anything here (besides the indentation), I just added `if` at the beginning of this method

**Comment by krzyk** [comment]
@paulodamaso corrected and replied regarding ClaimsRouting

**Comment by paulodamaso** [comment]
@krzyk That' good to me, let's not tamper with that `ClaimsRoutine` since you didn't changed it internal logic.

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by krzyk** [comment]
@g4s8 corrected, please take a look

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1578](https://github.com/zerocracy/farm/issues/1578) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**